thunder/distributed/transforms/fsdp_v2.py

```diff
@@ -212,6 +214,15 @@ def transform_module(
                 )
                 new_shape = thunder_model._overrides_parameters[pn].shape
                 self.sharded_params[pn] = (old_shape, new_shape, thunder_model._overrides_parameters[pn].device)
+                if self.release_original_parameters:
+                    base_pn = pn.rsplit(".", 1)[-1]
+                    p_orig = getattr(submodule, base_pn)
+                    if p_orig.device.type != "meta":
+                        p_meta = torch.nn.Parameter(p.to(device="meta"), requires_grad=p.requires_grad)
+                        p_meta._thunder_device = p_orig.device
+                        setattr(submodule, base_pn, p_meta)
```

Comment:
Can't we use [`torch.nn.Module.register_parameter`](https://pytorch.org/docs/stable/generated/torch.nn.Module.html#torch.nn.Module.register_parameter) or something like that, instead of `setattr`?

Reply:
We do know that it is a parameter, so setattr will do the right thing, but I'm happy to change it to register_parameter if you prefer.

Reply:
```suggestion
                        submodule.register_parameter(base_pn, p_meta)
```